WordPress/Classes/Models/Blog+Lookup.swift

```diff
@@ -69,6 +69,42 @@ public extension Blog {
         try? BlogQuery().xmlrpc(matching: xmlrpc).selfHostedBlogUsername(username).blog(in: context)
     }
 
+    /// Searches for a `Blog` object for this account with the given XML-RPC endpoint
+    ///
+    /// - Warning: If more than one blog is found, they'll be considered duplicates and be
+    /// deleted leaving only one of them.
+    ///
+    /// - Parameters:
+    ///   - xmlrpc: the XML-RPC endpoint URL as a string
+    ///   - account: the account the blog belongs to
+    ///   - context: the NSManagedObjectContext containing the account and the Blog object.
+    /// - Returns: the blog if one was found, otherwise it returns nil
+    static func lookup(xmlrpc: String, andRemoveDuplicateBlogsOf account: WPAccount, in context: NSManagedObjectContext) -> Blog? {
+        let predicate = NSPredicate(format: "xmlrpc like %@", xmlrpc)
+        let foundBlogs = account.blogs.filter { predicate.evaluate(with: $0) }
+
+        if foundBlogs.isEmpty {
+            return nil
+        }
+
+        if foundBlogs.count == 1 {
+            return foundBlogs.first
+        }
```

Comment:
Have you considered `guard` for these early returns?

```suggestion
        guard foundBlogs.isEmpty == false else {
            return nil
        }

         // It is fine to ignore the count == 0 case here because we have an isEmpty check above
        guard foundBlogs.count != 1 else {
            return foundBlogs.first
        }
```

However, that suggestion made me wonder whether it's acceptable to compute the count before starting the checks, even though some times (`isEmpty` true) it's a waste.

```suggestion
        let count = foundBlogs.count
        
        guard count != 0 else  {
            return nil
        }

        guard count != 1 else {
            return foundBlogs.first
        }
```

🤷 this is really gold plating anyways...

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/20047/commits/150f62bc624c76fa8f126494d0d5a5942e8a455c